diff --git a/dist/main.d.mts b/dist/main.d.mts
index b8a6b58e5bcdb97c0d8b0aaff372b90977bc2a92..0b40dbc6a00183cb828906d6ffb49af94adcc640 100644
--- a/dist/main.d.mts
+++ b/dist/main.d.mts
@@ -253,21 +253,26 @@ declare const connectSnap: (snapId?: string, params?: Record<'version', unknown>
 declare const getSnap: (snapId?: string, version?: string) => Promise<Snap | undefined>;
 declare const isLocalSnap: (snapId: string) => boolean;
 
+type ChainId = number;
+
 type SdkConfig = {
-    contracts: {
-        kycGuardianWhitelist: string;
-        zkKycRegistry: string;
-        verificationSbt: string;
-        exampleZkKycAgeProof: string;
-        exampleZkpVerifier: string;
-        exampleDapp: string;
-        exampleInstitution: string[];
-        exampleZkKyc: string;
-        repeatableZkpTest: string;
-        devnetGuardian: string;
-    };
-    defaultSnapOrigin: string;
-    defaultSnapVersion: string;
+  contracts: Record<
+    ChainId,
+    {
+      kycGuardianWhitelist: string;
+      zkKycRegistry: string;
+      verificationSbt: string;
+      exampleZkKycAgeProof: string;
+      exampleZkpVerifier: string;
+      exampleDapp: string;
+      exampleInstitution: string[];
+      exampleZkKyc: string;
+      repeatableZkpTest: string;
+      devnetGuardian: string;
+    }
+  >;
+  defaultSnapOrigin: string;
+  defaultSnapVersion: string;
 };
 declare const sdkConfig: SdkConfig;
 
diff --git a/dist/main.d.ts b/dist/main.d.ts
index b8a6b58e5bcdb97c0d8b0aaff372b90977bc2a92..95d06d4c0fcf7028e255bdcd16701585c01cb649 100644
--- a/dist/main.d.ts
+++ b/dist/main.d.ts
@@ -253,22 +253,28 @@ declare const connectSnap: (snapId?: string, params?: Record<'version', unknown>
 declare const getSnap: (snapId?: string, version?: string) => Promise<Snap | undefined>;
 declare const isLocalSnap: (snapId: string) => boolean;
 
+type ChainId = number;
+
 type SdkConfig = {
-    contracts: {
-        kycGuardianWhitelist: string;
-        zkKycRegistry: string;
-        verificationSbt: string;
-        exampleZkKycAgeProof: string;
-        exampleZkpVerifier: string;
-        exampleDapp: string;
-        exampleInstitution: string[];
-        exampleZkKyc: string;
-        repeatableZkpTest: string;
-        devnetGuardian: string;
-    };
-    defaultSnapOrigin: string;
-    defaultSnapVersion: string;
+  contracts: Record<
+    ChainId,
+    {
+      kycGuardianWhitelist: string;
+      zkKycRegistry: string;
+      verificationSbt: string;
+      exampleZkKycAgeProof: string;
+      exampleZkpVerifier: string;
+      exampleDapp: string;
+      exampleInstitution: string[];
+      exampleZkKyc: string;
+      repeatableZkpTest: string;
+      devnetGuardian: string;
+    }
+  >;
+  defaultSnapOrigin: string;
+  defaultSnapVersion: string;
 };
+
 declare const sdkConfig: SdkConfig;
 
 /**
diff --git a/dist/main.js b/dist/main.js
index 89fe8d8607e74768b6f8417f2c0800a6bb9e47c6..ef6ed94446eaad86923c84710534c664ab95f02b 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -88,24 +88,42 @@ var RpcResponseErr = /* @__PURE__ */ ((RpcResponseErr2) => {
 
 // src/config/index.ts
 var sdkConfig = {
+  defaultSnapOrigin: 'npm:@galactica-net/snap',
+  defaultSnapVersion: '0.6.1',
   contracts: {
-    kycGuardianWhitelist: "0xB95314E42d8Da05b1D805039F7b96e3935584543",
-    zkKycRegistry: "0x0415E990e55071F0d448F87CD170528C7783A484",
-    verificationSbt: "0x062DaB74A2709EC730FE11b3d9C00033B0FBAf92",
-    exampleZkKycAgeProof: "0x1554B2D7422941Be9903040d9946da60d9f2bC3C",
-    exampleZkpVerifier: "0xA437DFE87F9096e52E4dEcAA7cfc14d82A5aE07e",
-    exampleDapp: "0xBD2FAA4835E1A462Ea4FDA39352F24C58e237e2a",
-    exampleInstitution: [
-      "0x5E523B1c7c04eB971D3b851C82d3935b91fb4b3e",
-      "0xcc433436B6c01142EC9B95a1b80B3734CD4849f3",
-      "0x8d5ebACB647aea8a76082A36C93997765A50AF0E"
-    ],
-    exampleZkKyc: "0xD95efF72F06079DEcE33b18B165fc3A7a4bdc1fD",
-    repeatableZkpTest: "0xa99fcD678D985fB2ac8aD3fE913aED88705A44fc",
-    devnetGuardian: "0x71d80ea7744302E5b1cFD61a7a26153FF221ca9E"
+    '41238': {
+      kycGuardianWhitelist: '0xB95314E42d8Da05b1D805039F7b96e3935584543',
+      zkKycRegistry: '0x0415E990e55071F0d448F87CD170528C7783A484',
+      verificationSbt: '0x062DaB74A2709EC730FE11b3d9C00033B0FBAf92',
+      exampleZkKycAgeProof: '0x1554B2D7422941Be9903040d9946da60d9f2bC3C',
+      exampleZkpVerifier: '0xA437DFE87F9096e52E4dEcAA7cfc14d82A5aE07e',
+      exampleDapp: '0xBD2FAA4835E1A462Ea4FDA39352F24C58e237e2a',
+      exampleInstitution: [
+        '0x5E523B1c7c04eB971D3b851C82d3935b91fb4b3e',
+        '0xcc433436B6c01142EC9B95a1b80B3734CD4849f3',
+        '0x8d5ebACB647aea8a76082A36C93997765A50AF0E',
+      ],
+      exampleZkKyc: '0xD95efF72F06079DEcE33b18B165fc3A7a4bdc1fD',
+      repeatableZkpTest: '0xa99fcD678D985fB2ac8aD3fE913aED88705A44fc',
+      devnetGuardian: '0x71d80ea7744302E5b1cFD61a7a26153FF221ca9E',
+    },
+    '9301': {
+      kycGuardianWhitelist: '0x2547a64b4CDe45ad691BC393aBe20D405cD5139D',
+      zkKycRegistry: '0x8b7f9322F2CF92908eDB02a76DD8A2cAd6E566B5',
+      verificationSbt: '0x5A040bA7C0eA4ae51b07Fd247a16BA767D716A75',
+      exampleZkKycAgeProof: '0xdA651B29318Ca44d5f99DFA102483fBF8ab6Fd32',
+      exampleZkpVerifier: '0x538209Ca8ccA04e1548995B0F89B1f829718c2e0',
+      exampleDapp: '0x249c63682a93e4044467eD4C4EABF2D010425735',
+      exampleInstitution: [
+        '0x6f318e025977BEb146CBa6376cc642ffe142EB00',
+        '0x6E5eeD7b04532C33E3689aEFd5596E98Cb694bF5',
+        '0xc05e8FA7FB5Ff3c8F9fAe95a62c751498EFFF3E7',
+      ],
+      exampleZkKyc: '0x200aE9f8D04c152128D78856E3e95d200915F69E',
+      repeatableZkpTest: '0x099c91224A1D07a3F63E37F759797cF59D121534',
+      devnetGuardian: '0x3AFF448DdF6F75d3D92DC46A47e230181d174614',
+    },
   },
-  defaultSnapOrigin: "npm:@galactica-net/snap",
-  defaultSnapVersion: "0.6.1"
 };
 
 // src/utils/invoke-snap.ts
diff --git a/dist/main.mjs b/dist/main.mjs
index e2105650c1a1c24c077e6956f5e2a37e26fed17b..7d7c308e7288fc00198df271cb0cf241b793a220 100644
--- a/dist/main.mjs
+++ b/dist/main.mjs
@@ -42,24 +42,42 @@ var RpcResponseErr = /* @__PURE__ */ ((RpcResponseErr2) => {
 
 // src/config/index.ts
 var sdkConfig = {
+  defaultSnapOrigin: 'npm:@galactica-net/snap',
+  defaultSnapVersion: '0.6.1',
   contracts: {
-    kycGuardianWhitelist: "0xB95314E42d8Da05b1D805039F7b96e3935584543",
-    zkKycRegistry: "0x0415E990e55071F0d448F87CD170528C7783A484",
-    verificationSbt: "0x062DaB74A2709EC730FE11b3d9C00033B0FBAf92",
-    exampleZkKycAgeProof: "0x1554B2D7422941Be9903040d9946da60d9f2bC3C",
-    exampleZkpVerifier: "0xA437DFE87F9096e52E4dEcAA7cfc14d82A5aE07e",
-    exampleDapp: "0xBD2FAA4835E1A462Ea4FDA39352F24C58e237e2a",
-    exampleInstitution: [
-      "0x5E523B1c7c04eB971D3b851C82d3935b91fb4b3e",
-      "0xcc433436B6c01142EC9B95a1b80B3734CD4849f3",
-      "0x8d5ebACB647aea8a76082A36C93997765A50AF0E"
-    ],
-    exampleZkKyc: "0xD95efF72F06079DEcE33b18B165fc3A7a4bdc1fD",
-    repeatableZkpTest: "0xa99fcD678D985fB2ac8aD3fE913aED88705A44fc",
-    devnetGuardian: "0x71d80ea7744302E5b1cFD61a7a26153FF221ca9E"
+    '41238': {
+      kycGuardianWhitelist: '0xB95314E42d8Da05b1D805039F7b96e3935584543',
+      zkKycRegistry: '0x0415E990e55071F0d448F87CD170528C7783A484',
+      verificationSbt: '0x062DaB74A2709EC730FE11b3d9C00033B0FBAf92',
+      exampleZkKycAgeProof: '0x1554B2D7422941Be9903040d9946da60d9f2bC3C',
+      exampleZkpVerifier: '0xA437DFE87F9096e52E4dEcAA7cfc14d82A5aE07e',
+      exampleDapp: '0xBD2FAA4835E1A462Ea4FDA39352F24C58e237e2a',
+      exampleInstitution: [
+        '0x5E523B1c7c04eB971D3b851C82d3935b91fb4b3e',
+        '0xcc433436B6c01142EC9B95a1b80B3734CD4849f3',
+        '0x8d5ebACB647aea8a76082A36C93997765A50AF0E',
+      ],
+      exampleZkKyc: '0xD95efF72F06079DEcE33b18B165fc3A7a4bdc1fD',
+      repeatableZkpTest: '0xa99fcD678D985fB2ac8aD3fE913aED88705A44fc',
+      devnetGuardian: '0x71d80ea7744302E5b1cFD61a7a26153FF221ca9E',
+    },
+    '9301': {
+      kycGuardianWhitelist: '0x2547a64b4CDe45ad691BC393aBe20D405cD5139D',
+      zkKycRegistry: '0x8b7f9322F2CF92908eDB02a76DD8A2cAd6E566B5',
+      verificationSbt: '0x5A040bA7C0eA4ae51b07Fd247a16BA767D716A75',
+      exampleZkKycAgeProof: '0xdA651B29318Ca44d5f99DFA102483fBF8ab6Fd32',
+      exampleZkpVerifier: '0x538209Ca8ccA04e1548995B0F89B1f829718c2e0',
+      exampleDapp: '0x249c63682a93e4044467eD4C4EABF2D010425735',
+      exampleInstitution: [
+        '0x6f318e025977BEb146CBa6376cc642ffe142EB00',
+        '0x6E5eeD7b04532C33E3689aEFd5596E98Cb694bF5',
+        '0xc05e8FA7FB5Ff3c8F9fAe95a62c751498EFFF3E7',
+      ],
+      exampleZkKyc: '0x200aE9f8D04c152128D78856E3e95d200915F69E',
+      repeatableZkpTest: '0x099c91224A1D07a3F63E37F759797cF59D121534',
+      devnetGuardian: '0x3AFF448DdF6F75d3D92DC46A47e230181d174614',
+    },
   },
-  defaultSnapOrigin: "npm:@galactica-net/snap",
-  defaultSnapVersion: "0.6.1"
 };
 
 // src/utils/invoke-snap.ts
